app/assets/javascripts/components/materials_collection/materials_collection.js.coffee

```diff
@@ -0,0 +1,45 @@
+{div, span, a} = React.DOM
+
+window.MaterialsCollectionClass = React.createClass
+  getInitialState: ->
+    materials: []
+    truncated: true
+
+  componentDidMount: ->
+    jQuery.ajax
+      url: Portal.API_V1.MATERIALS_COLLECTION_DATA
+      data: id: @props.collection
+      dataType: 'json'
+      success: (data) =>
+        @setState materials: data[0].materials if @isMounted()
+
+  toggle: (e) ->
+    @setState truncated: not @state.truncated
+    e.preventDefault()
+
+  getMaterialsList: ->
+    if @state.truncated
+      @state.materials.slice 0, @props.limit
+    else
+      @state.materials
+
+  renderTruncationToggle: ->
+    (a {className: 'mc-truncate', onClick: @toggle, href: ''},
+      (span {},
```

Comment:
You can skip this span - React will render the plain text in the if/else as spans automatically.